src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine_FindReferencesInDocuments.cs

```diff
@@ -110,19 +112,29 @@ async ValueTask PerformSearchInDocumentWorkerAsync(ISymbol symbol, FindReference
 
         async ValueTask DirectSymbolSearchAsync(ISymbol symbol, FindReferencesDocumentState state)
         {
-            using var _ = ArrayBuilder<FinderLocation>.GetInstance(out var referencesForFinder);
+            var channel = Channel.CreateUnbounded<FinderLocation>();
```

Comment:
>Channel.CreateUnbounded [](http://example.com/codeflow?start=26&length=23)

I've become accustomed to the ProduceConsumer calls, did this not fit into that model?

Reply:
updated!